### ISSUE TITLE
Firefox tabs take 15-20 sec to start loading after initial launch

### ISSUE THREAD

**Issue by xDarksome** [pop-os/cosmic-comp#102]
My workflow:
- Start `cosmic-comp` from a tty
- Launch `firefox`
- Assume `firefox` restores previously opened tabs
- Tabs are being opened but the content starts loading with a delay of 15-20 sec

The issue does not persist upon reopening `firefox`. Also, the issue doesn't seem to exist while launching `cosmic-comp` from another compositor (`cosmic-comp` inside another `cosmic-comp` or `cosmic-comp` inside `hyprland`)
Looks like something related to hardware acceleration. 

Any clues on where to dig?

**Comment by Vixea** [comment]
It sounds like this issue and #104 are the same thing no need to create two issues for actually only one

**Comment by xDarksome** [comment]
I'm not sure they are the same. They may have the same root cause, maybe not. 
Subsequent launches of the compositor do not have this initial lag, subsequent launches of Firefox do.

**Comment by Drakulix** [comment]
Right, so just using `dbus-run-session cosmic-comp` might already fix a bunch of stuff for you.

E.g. our `start-cosmic` script might be of much use for you to setup the environment (or at least be a good reference to try and build your own): https://github.com/pop-os/cosmic-session/blob/master/data/start-cosmic

Just replace the occurrences of the `cosmic-session` binary with `cosmic-comp`.

**Comment by xDarksome** [comment]
I've changed the script to the following one:
```bash

#!/bin/bash

# From: https://people.debian.org/~mpitt/systemd.conf-2016-graphical-session.pdf

# robustness: if the previous graphical session left some failed units,
# reset them so that they don't break this startup
for unit in $(systemctl --user --no-legend --state=failed --plain list-units | cut -f1 -d' '); do
    partof="$(systemctl --user show -p PartOf --value "$unit")"
    for target in graphical-session.target; do
        if [ "$partof" = "$target" ]; then
            systemctl --user reset-failed "$unit"
            break
        fi
    done
done

# /etc/profile contains a lot of important environment variables
source /etc/profile

# load user's profile if it exists
if [ -n "${HOME}" -a -f "${HOME}/.profile" ]
then
    source "${HOME}/.profile"
fi

export XDG_CURRENT_DESKTOP="${XDG_CURRENT_DESKTOP:=pop:COSMIC}"
export XDG_SESSION_TYPE="${XDG_SESSION_TYPE:=wayland}"
export XCURSOR_THEME="${XCURSOR_THEME:=Pop}"
export _JAVA_AWT_WM_NONREPARENTING=1
export GDK_BACKEND=wayland,x11
export MOZ_ENABLE_WAYLAND=1

# import environment variables from the login manager
systemctl --user import-environment XDG_SESSION_TYPE XDG_CURRENT_DESKTOP

# Run cosmic-session
if [[ -z "${DBUS_SESSION_BUS_ADDRESS}" ]]; then
    exec dbus-run-session -- cosmic-comp
else
    exec cosmic-comp
fi
````

Basically, removed `cosmic-session.target` service and removed the full paths to binaries (as they are not the same for me, I'm on NixOS). And the only this that's changed is the size and style of the mouse cursor :smile: 

The issue persists.

**Comment by Drakulix** [comment]
Cosmic-session per se shouldn't be required, but it might launch components, that firefox expects to be available.

Debugging firefox to figure out, what is causing the timeout might be the better solution though. I know the MOZ_LOG environment variable is pretty powerful to do so, but I don't know what you would be specifically looking for unfortunately.

**Comment by xDarksome** [comment]
So, I didn't manage to make firefox to access MOZ_LOG for some reason (most likely because the version under NixOS which I use is wrapped with a bash script, but whatever).

Enabling `--safe-mode` fixed the issue, then I started disabling extensions one by one, and eventually found the guilty one -  Temporary Containers. 
In the whole visible universe there is no adequate reason for changing wayland compossitor to have an effect on this. :upside_down_face: 

I'll try to dig dipper tomorrow

**Comment by xDarksome** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1734947

It seems that the root cause of the issue hasn't been identified to this day. The last comment also suggest there may be some correlation with graphics.

> Hello! I discovered some information that may be helpful
I can reliably reproduce the startup timeout/delay with temporary containers installed when I set my monitor's refresh rate to 300hz
The delay does not occur at 240hz or 60hz

`Hyprland` doesn't have this issue on both 240 and 60 hz 

I'll try to play with different refresh rate option on `cosmic-comp`

**Comment by jokeyrhyme** [comment]
Probably https://github.com/flatpak/xdg-desktop-portal/issues/1032 or https://github.com/swaywm/sway/wiki#gtk-applications-take-20-seconds-to-start

To start with, try completely uninstalling xdg-desktop-portal-gnome before starting COSMIC (or any other compositor) and only install it before starting GNOME

**Comment by xDarksome** [comment]
@jokeyrhyme I think the problem you're referring is this one https://github.com/pop-os/cosmic-comp/issues/104 . 
I don't have `xdg-desktop-portal` installed currently, or any other compositor.

Thanks for suggestion anyway!
